docs/deprecation.md

```diff
@@ -0,0 +1,23 @@
+# Deprecation Policy
+
+## User-facing commands and their arguments
+
+Whenever you rename a command or make a previously optional argument required,
+we require that you preserve the old command invocations keeps on working for 6
+months (so 6 releases, since we release monthly) with a deprecation message.
+The message should inform the user that the previous workflow is deprecated
+and to be removed in the future.
+
+## Packaging commands and niche commands
+
+For commands with a niche user audience or something we assume is rarely used
+(we sadly have no data), we take the liberity to remove the old behavior within
+two releases. This means that you can change the old command to immediately
+error.
```

Comment:
What do you mean about "within two releases"? This seems to contradict making the command to immediately error.

Reply:
I've added a example with a debug command